src/grid/interpolate.jl

```diff
@@ -21,6 +21,83 @@ InterpStyle(::Type) = FloorInterp()
 InterpStyle(::Type{<:SimpleGrid.BaryCheb}) = ChebInterp()
 InterpStyle(::Type{<:CompositeGrid.Composite}) = CompositeInterp()
 
+"""
+    function linearND(data, xgrids, xs)
+
+linear interpolation of data(xs)
+
+#Arguments:
+- xgrids: n-dimensional grids, xgrids[i] is a 1D grid
+- data: n-dimensional array of data
+- xs: list of x, x[i] corresponds to xgrids[i]
+"""
+function linearND(data, xgrids, xs)
+    function enumX(xs)
```

Comment:
interpolation needs to be very fast. Function in function can be dangerous for performance.